api/v3/Job.php

```diff
@@ -470,7 +470,14 @@ function civicrm_api3_job_process_membership($params) {
     return civicrm_api3_create_error('Could not acquire lock, another Membership Processing process is running');
   }
 
-  $result = CRM_Member_BAO_Membership::updateAllMembershipStatus();
+  // We need to pass this through as a simple array of membership status IDs as values.
```

Comment:
If the purpose of this code is to accept variations in peoples preferences for passing parameters, I'm not sure I would be so forgiving. Instead I would expect people to pass in parameters in the correct format.

Reply:
@mattwire I agree with @michaelmcandrew on this - but the presence of many tests was pushing me to merge it, even though this chuck of code doesn't really make sense to me 

However, I am concerned that we don't declare a type on exclude_membership_status_ids in the metadata and we are not validating it at the api layer & the validation at the next layer is a bit unclear too.

I think that if it were declare as a INT with a pseudoconstant then an array would be accepted but it would be validated

Reply:
@eileenmcnaughton @michaelmcandrew I've added the metadata for the `exclude_membership_status_ids` parameter - it's now validating as an INT but I don't get a pseudoconstant selector in the API3 UI.